service/index/writer.go

```diff
@@ -228,7 +228,12 @@ func (w *Writer) apply(ops ...func(*badger.Txn) error) error {
 	// previous transaction.
 	select {
 	case err := <-w.err:
-		return fmt.Errorf("could not commit transaction: %w", err)
+		// Only return an error if the error channel received an error, not if it
+		// closed unexpectedly.
+		if err != nil {
+			return fmt.Errorf("could not commit transaction: %w", err)
+		}
+		return nil
```

Comment:
This is unnecessary. The error channel is only ever closed at the end of `Close()`, at which point all of the transactions have already committed. We never send anything but a real error on this. Or did you run into a case where this wasn't true?

Reply:
Well when we close it, we get a `nil` error here, so we dont want to return an error if it closed naturally, do we? 🤔

Reply:
`apply()` is never called after we reach `close(w.err)` in the `Close()` function, that's the point of my comment.

Reply:
Theoretically, though it's very unlikely in terms of timing, it's possible for an `apply` call to start in a goroutine while another goroutine is closing the writer, and for `w.err` to be closed before we reach the `select` in the `apply` function. Or at least there's no guarantee that it can't happen.

So yeah I can change it to properly check for a closed channel then, it's true that it's cleaner this way than to just check whether we get a `nil` error.

Reply:
No, this is not possible. `committed` is only called by asynchronous transaction commits. In `Close()`, we wait for _all_ of them to finish before closing the error channel.